File: VB/FilterDuplicateRecords_UnboundEditable/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.Xpf
Imports System
Imports System.Collections.ObjectModel

Namespace FilterDuplicateRecords_UnboundEditable

    Public Class MainViewModel
        Inherits ViewModelBase

        Public ReadOnly Property Source As ObservableCollection(Of Item) = New ObservableCollection(Of Item)(Item.GetData(1000))

        <Command>
        Public Sub UnboundColumnDataCommand(ByVal e As UnboundColumnRowArgs)
            If Equals(e.FieldName, "GrowthUnbound") Then
                If e.IsGetData Then
                    e.Value = Math.Round(Source(e.SourceIndex).Growth, 2)
                    Return
                End If

                If e.IsSetData AndAlso TypeOf e.Value Is Decimal Then
                    Source(e.SourceIndex).Growth = CDec(e.Value)
                End If
            End If
        End Sub
    End Class
End Namespace

File: CS/FilterDuplicateRecords_UnboundEditable/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.Xpf;
using System;
using System.Collections.ObjectModel;

namespace FilterDuplicateRecords_UnboundEditable
{
    public class MainViewModel : ViewModelBase
    {
        public ObservableCollection<Item> Source { get; } = new ObservableCollection<Item>(Item.GetData(1000));

        [Command]
        public void UnboundColumnDataCommand(UnboundColumnRowArgs e) {
            if (e.FieldName == "GrowthUnbound") {
                if (e.IsGetData) {
                    e.Value = Math.Round(Source[e.SourceIndex].Growth, 2);
                    return;
                }

                if (e.IsSetData && e.Value is decimal number) {
                    Source[e.SourceIndex].Growth = number;
                }
            }
        }
    }
}
